### ISSUE TITLE
Compilation error in image_filter.hpp

### ISSUE THREAD

**Issue by pnorman** [mapnik/mapnik#2859]
With clang 3.4.1

```
In file included from src/save_map.cpp:40:
In file included from include/mapnik/image_filter.hpp:35:
In file included from /usr/local/include/boost/gil/gil_all.hpp:26:
/usr/local/include/boost/gil/channel_algorithm.hpp:54:85: error: non-type template argument evaluates to -1, which cannot be narrowed to type 'unsigned long'
      [-Wc++11-narrowing]
struct unsigned_integral_max_value : public mpl::integral_c<UnsignedIntegralChannel,-1> {};
                                                                                    ^
/usr/local/include/boost/gil/channel_algorithm.hpp:204:19: note: in instantiation of template class 'boost::gil::detail::unsigned_integral_max_value<unsigned
      long>' requested here
        if (src > unsigned_integral_max_value<uintmax_t>::value - div2)
```

This does not occur with clang 3.6.0, but other errors do.

**Comment by flippmoke** [comment]
What version of boost are you using?

**Comment by flippmoke** [comment]
I think this is because of the 
```
        if 'clang++' in env['CXX']:
            #  - workaround boost gil channel_algorithm.hpp narrowing error
            # TODO - remove when building against >= 1.55
            # https://github.com/mapnik/mapnik/issues/1970
            env.Append(CXXFLAGS = '-Wno-c++11-narrowing')
```
issue.

**Comment by pnorman** [comment]
> What version of boost are you using?

1.55

**Comment by pnorman** [comment]
If I force that check by using `clang++` instead of `c++`, it builds, even though they are the same compiler.

So it looks like the TODO is wrong, and the check should be replaced by one that checks what compiler it is.

**Comment by springmeyer** [comment]
fixed in 4667e94
